src/program/services/streaming/media_stream.py

```diff
@@ -126,12 +132,34 @@ def __init__(
                 )
             )
 
-        try:
-            self.async_client = di[httpx.AsyncClient]
-        except KeyError:
-            raise RuntimeError(
-                "httpx.AsyncClient not found in dependency injector"
-            ) from None
+        # Use proxy for AllDebrid if configured
+        if provider == "alldebrid":
+            proxy_url = settings_manager.settings.downloaders.proxy_url
+            if proxy_url:
+                logger.log(
+                    "STREAM",
+                    self._build_log_message(
+                        f"Using proxy {proxy_url} for AllDebrid streaming"
+                    ),
+                )
+                # Create a dedicated client with proxy for this stream
+                mounts = {
+                    "http://": httpx.AsyncHTTPTransport(proxy=proxy_url),
+                    "https://": httpx.AsyncHTTPTransport(proxy=proxy_url),
+                }
+                self.async_client = httpx.AsyncClient(http2=True, mounts=mounts)
```

Comment:
```suggestion
                self.async_client = httpx.AsyncClient(http2=True, proxy=proxy_url)
```

https://www.python-httpx.org/troubleshooting/#proxies

> Change the scheme of your HTTPS proxy to http://... instead of https://...:

```
mounts = {
  "http://": httpx.HTTPTransport(proxy="http://myproxy.org"),
  "https://": httpx.HTTPTransport(proxy="http://myproxy.org"),
}
```

> This can be simplified to:

```
proxy = "http://myproxy.org"
with httpx.Client(proxy=proxy) as client:
```